### ISSUE TITLE
Incorrect ALTER TABLE statement generated in Schema sync with 'INT' datatype

### ISSUE THREAD

**Issue by sibwy** [webyog/sqlyog-community#2263]
The schema synchronization tool returns a schema mismatch on comparing two similar tables. The reason is MySQL v8.0+ does not use INT(10), but only INT for the data type.

Issue reproducible with the following table created in source and target (In the target DB, the INT datatype in the table 't5' should be defined as "INT (10)" and in 't6', just "INT" and no width to be defined.),

```
CREATE TABLE `t5` (
`id` INT UNSIGNED NOT NULL AUTO_INCREMENT COMMENT 'ID',
`name` VARCHAR(10) NOT NULL COMMENT 'name',
  PRIMARY KEY (`id`),
  UNIQUE KEY `unique_name` (`name`)
) ENGINE=INNODB AUTO_INCREMENT=1 DEFAULT CHARSET=latin1;
```
```
CREATE TABLE `t6` (
`id` INT(10) UNSIGNED NOT NULL AUTO_INCREMENT COMMENT 'ID',
`name` VARCHAR(10) NOT NULL COMMENT 'name',
  PRIMARY KEY (`id`),
  UNIQUE KEY `unique_name` (`name`)
) ENGINE=INNODB AUTO_INCREMENT=1 DEFAULT CHARSET=latin1;
```
This issue will also not let the user perform the data sync since the tool reports a mismatch with the schema.

The user reports can be found in SF >> 00815445 and 00815463

**Comment by sibwy** [comment]
The reason behind the issue is,

MySQL does not display the default INTEGER datatype width in the datatype definition as it was deprecated in v8.0.17 and the width of the INT datatype is not displayed in the datatype definition in v8.0.19 and above. 

For more details, [here](https://dev.mysql.com/doc/refman/8.0/en/numeric-type-syntax.html#:~:text=As of MySQL 8.0.17, (and any synonyms); you should expect support for it to be removed in a future version of MySQL.).

**Comment by sibwy** [comment]
Fixed in [v13.1.8](https://blog.sqlyog.com/sqlyog-mysql-gui-13-1-8-release/).

> SQLyog generated an incorrect ALTER TABLE statement in Schema synchronization with ‘TIMESTAMP’ and integer (INT, TINYINT, BIGINT, and SMALLINT) data types for MySQL Server version 8.0.19 or above with any target MySQL version and vice versa. Though the Schema Sync will still display the difference in the schema, we now allow the data to be synced for only INT, TINYINT, BIGINT, and SMALLINT data types.

**Comment by francoisstark400** [comment]
My problem is with TINTYINT (1).  
- Mysql does not have BOOL, it uses TINYINT(1) as the equivalent. (https://dev.mysql.com/doc/refman/8.0/en/numeric-type-syntax.html)
- So how are we supposed to define TINYINT (1) columns if INT types' display width is deprecated?
- We will not be able to read INT display width from mysql 8.0.19: https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-19.html  -  "**now statements that include data type definitions in their output no longer show the display width for integer types**"

- BUT right below it, on the 8.0.19 news page, it says there is an exception: TINYINT(1):  **The type is TINYINT(1). MySQL Connectors make the assumption that TINYINT(1) columns originated as BOOLEAN columns; this exception enables them to continue to make that assumption.**

Which means that the SQLYog Schema Synchronization tool must ALSO make that exception please - You will have to catch and allow TINYINT (1) to read and write the display with.
